dummyserver/hypercornserver.py

```diff
@@ -42,8 +44,20 @@ async def _start_server(
 
 @contextlib.contextmanager
 def run_hypercorn_in_thread(
-    config: hypercorn.Config, app: hypercorn.typing.ASGIFramework
-) -> Generator[None, None, None]:
+    host: str, certs: dict[str, typing.Any] | None, app: hypercorn.typing.ASGIFramework
+) -> typing.Iterator[int]:
```

Comment:
Iterator is technically the wrong type, because contextmanager incorrectly accepts an object without `.throw`

```suggestion
) -> Generator[int, None, None]
```

Reply:
Since this is accepted by mypy, could be fixed in typeshed, is only in our tests and is only "technically wrong", I'd rather keep it as it's easier to understand that way.